### PR TITLE
feat: SortOrder methods should take schema ref if possible

### PR DESCRIPTION
**Pull request by c-thiel**
Reduce clones

I also added two tests that didn't fail but where missing IMO

### CODE REVIEW

File: crates/iceberg/src/spec/sort.rs [hunk 1]
```diff
@@ -133,6 +133,14 @@ impl SortOrder {
     pub fn is_unsorted(&self) -> bool {
         self.fields.is_empty()
     }
+
+    /// Set the order id for the sort order
+    pub fn with_order_id(&self, order_id: i64) -> SortOrder {
```
Comment (Xuanwo) [inline crates/iceberg/src/spec/sort.rs:138]: An API named `with_xxx` that takes a ref might be a bit confusing. Is there any context for this?

**Review by Xuanwo — COMMENTED** [review]
Others LGTM, only one small question.

File: crates/iceberg/src/spec/sort.rs (hunk 1 above)
Comment (c-thiel) [reply crates/iceberg/src/spec/sort.rs]: You are right, owned version is much cleaner. Fixed in latest commit :)

**Review by Xuanwo — APPROVED** [review]
Thank you for this change.

**Pull request merged**